### PR TITLE
ci: timed check for dependency updates

### PR DESCRIPTION
**Pull request by Wxh16144**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [x] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- #45579
- #45453
- #45330
- #45228 
...

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    --       |
| 🇨🇳 Chinese |     --      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at ff7a910</samp>

This pull request adds a workflow to automatically check and update the ant-design and react-component dependencies in the `package.json` file and create a pull request with the changes. It also adds a configuration file for the `npm-check-updates` tool that defines the filter and options for the dependency upgrade.

### <samp>🤖 Generated by Copilot at ff7a910</samp>

* Add a new workflow to check and update dependencies every day or manually ([link](https://github.com/ant-design/ant-design/pull/45593/files?diff=unified&w=0#diff-a1ddd58555b460f24a369233e83d58cb9433fc429c77bb09e094f039df6bbceaR1-R61))
* Add a new configuration file for `npm-check-updates` tool ([link](https://github.com/ant-design/ant-design/pull/45593/files?diff=unified&w=0#diff-c0375955ab623100b75314536829e7f50637cab5d1539b2f2b8f7f9206b01440R1-R15))

### CODE REVIEW

File: .github/workflows/upgrade-deps.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            echo "upgrade-deps<<EOF" >> $GITHUB_OUTPUT
+            cat .tmp/upgrade-deps.txt >> $GITHUB_OUTPUT
+            echo "EOF" >> $GITHUB_OUTPUT
+          fi
+
+      - name: create pull request
+        id: cpr
+        uses: peter-evans/create-pull-request@v5
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }} # Cannot be default!!!
```
Comment (Wxh16144) [inline .github/workflows/upgrade-deps.yml:48]: 这里不能直接使用 `{{ secrets.GITHUB_TOKEN }}`, 因为它创建的 PR 不会触发 pull-request 流水线，需要配置一个

详见文档： https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#triggering-further-workflow-runs



CC @afc163

**Comment by afc163** [comment]
应该不需要自己实现，直接配置 https://github.com/ant-design/ant-design/blob/master/.github/dependabot.yml 就行。dependabot 本身也支持了 group update：https://github.com/dependabot/dependabot-core/issues/1190#issuecomment-1693360104

**Comment by Wxh16144** [comment]
> 应该不需要自己实现，直接配置 https://github.com/ant-design/ant-design/blob/master/.github/dependabot.yml 就行。dependabot 本身也支持了 group update：[dependabot/dependabot-core#1190 (comment)](https://github.com/dependabot/dependabot-core/issues/1190#issuecomment-1693360104)

大意了，重复劳动了

**Comment by yoyo837** [comment]
需求是:
- [ ] 只更新 `master` 的依赖;
- [ ] 只更新 `@rc-component/*` 和 `rc-*` 的包空间;
- [ ] 只更新 `dependencies` 下的那些包;
- [ ] 对于 `~` 依赖, 只更新到当前 minor 的最新 patch,  对于`^` 依赖始终更新到最新;

**Comment by Wxh16144** [comment]
> 需要同时满足如下需求:
> 
> * [x]  只更新 `master` 的依赖;
> * [x]  只更新 `@rc-component/*` 和 `rc-*` 的包空间;
> * [x]  只更新 `dependencies` 下的那些包;
> * [x]  对于 `~` 依赖, 只更新到当前 minor 的最新 patch,  对于`^` 依赖始终更新到最新;

上述需求严格满足，并且 `@ant-design/` 组织的包也严格遵循 [semver](https://semver.org/) 更新策略

至于上面提到的 dependabot 没有深入研究 （略过）~

准备好了，随时 review~

File: .ncurc.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// doc: https://github.com/raineorshine/npm-check-updates/tree/v16.14.6#readme
```
Comment (Wxh16144) [inline .ncurc.js:1]: 这里保留了 npm-check-updates 的版本信息，所以没有在 package 显式添加依赖，后续如果有问题直接 lock 版本也行。（主要是 ci 里面就不需要 cache node_modules 直接 npx 执行即可

**Review by afc163 — APPROVED** [review]

**Pull request merged**